### DISCUSSION TITLE
Permission Controller crashes when trying to edit permissions

### DISCUSSION THREAD

**Discussion by Riahuoz** [community/community#112760] [Projects and Issues]
### Select Topic Area

Bug

### Feature Area

Issues

### Body

Device Samsung M31 | M315F
ROM  AfterLife 6.O
Whenever I try to access to Permissions It craches .. this is the Log that i got

  | time: 1710561184203
-- | --
  | msg: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.ImageView.setVisibility(int)' on a null object reference
  | stacktrace: android.view.InflateException: Binary XML file line #17 in com.android.permissioncontroller:layout/preference_category_settings: Attempt to invoke virtual method 'void android.widget.ImageView.setVisibility(int)' on a null object reference
  | Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.ImageView.setVisibility(int)' on a null object reference
  | at declan.prjct.settings.widget.DeclanCategoryLayout.updateIconView(DeclanCategoryLayout.java:108)
  | at declan.prjct.settings.widget.DeclanCategoryLayout.updateSettings(DeclanCategoryLayout.java:73)
  | at declan.prjct.settings.widget.DeclanCategoryLayout.onFinishInflate(DeclanCategoryLayout.java:53)
  | at android.view.LayoutInflater.rInflate(LayoutInflater.java:1148)
  | at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:1096)
  | at android.view.LayoutInflater.inflate(LayoutInflater.java:686)
  | at android.view.LayoutInflater.inflate(LayoutInflater.java:538)
  | at androidx.preference.PreferenceGroupAdapter.onCreateViewHolder(PreferenceGroupAdapter.java:399)
  | at androidx.preference.PreferenceGroupAdapter.onCreateViewHolder(PreferenceGroupAdapter.java:47)
  | at androidx.recyclerview.widget.RecyclerView$Adapter.createViewHolder(RecyclerView.java:7461)
  | at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6582)
  | at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6466)
  | at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6462)
  | at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2330)
  | at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1631)
  | at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1591)
  | at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:668)
  | at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4464)
  | at androidx.recyclerview.widget.RecyclerView.onMeasure(RecyclerView.java:3841)
  | at android.view.View.measure(View.java:26496)
  | at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6981)
  | at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
  | at android.view.View.measure(View.java:26496)
  | at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6981)
  | at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1608)
  | at android.widget.LinearLayout.measureVertical(LinearLayout.java:878)
  | at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
  | at android.view.View.measure(View.java:26496)
  | at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6981)
  | at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
  | at android.view.View.measure(View.java:26496)
  | at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6981)
  | at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
  | at android.view.View.measure(View.java:26496)
  | at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6981)
  | at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1608)
  | at android.widget.LinearLayout.measureVertical(LinearLayout.java:878)
  | at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
  | at android.view.View.measure(View.java:26496)
  | at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6981)
  | at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
  | at android.view.View.measure(View.java:26496)
  | at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6981)
  | at androidx.coordinatorlayout.widget.CoordinatorLayout.onMeasureChild(CoordinatorLayout.java:795)
  | at com.google.android.material.appbar.HeaderScrollingViewBehavior.onMeasureChild(HeaderScrollingViewBehavior.java:99)
  | at com.google.android.material.appbar.AppBarLayout$ScrollingViewBehavior.onMeasureChild(AppBarLayout.java:2253)
  | at androidx.coordinatorlayout.widget.CoordinatorLayout.onMeasure(CoordinatorLayout.java:866)
  | at android.view.View.measure(View.java:26496)
  | at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6981)
  | at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
  | at android.view.View.measure(View.java:26496)
  | at androidx.constraintlayout.widget.ConstraintLayout$Measurer.measure(ConstraintLayout.java:930)
  | at androidx.constraintlayout.core.widgets.analyzer.BasicMeasure.measure(BasicMeasure.java:491)
  | at androidx.constraintlayout.core.widgets.analyzer.BasicMeasure.measureChildren(BasicMeasure.java:140)
  | at androidx.constraintlayout.core.widgets.analyzer.BasicMeasure.solverMeasure(BasicMeasure.java:285)
  | at androidx.constraintlayout.core.widgets.ConstraintWidgetContainer.measure(ConstraintWidgetContainer.java:119)
  | at androidx.constraintlayout.widget.ConstraintLayout.resolveSystem(ConstraintLayout.java:1739)
  | at androidx.constraintlayout.widget.ConstraintLayout.onMeasure(ConstraintLayout.java:1882)
  | at android.view.View.measure(View.java:26496)
  | at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6981)
  | at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
  | at android.view.View.measure(View.java:26496)
  | at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6981)
  | at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1608)
  | at android.widget.LinearLayout.measureVertical(LinearLayout.java:878)
  | at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
  | at android.view.View.measure(View.java:26496)
  | at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6981)
  | at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
  | at com.android.internal.policy.DecorView.onMeasure(DecorView.java:760)
  | at android.view.View.measure(View.java:26496)
  | at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:4042)
  | at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:2658)
  | at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2964)
  | at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2371)
  | at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:9308)
  | at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1231)
  | at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1239)
  | at android.view.Choreographer.doCallbacks(Choreographer.java:899)
  | at android.view.Choreographer.doFrame(Choreographer.java:832)
  | at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1214)
  | at android.os.Handler.handleCallback(Handler.java:942)
  | at android.os.Handler.dispatchMessage(Handler.java:99)
  | at android.os.Looper.loopOnce(Looper.java:201)
  | at android.os.Looper.loop(Looper.java:288)
  | at android.app.ActivityThread.main(ActivityThread.java:7965)
  | at java.lang.reflect.Method.invoke(Native Method)
  | at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
  | at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:849)
 

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by queenofcorgis** [comment]
Hi @Riahuoz , Thanks for reaching out! 

We're sorry to hear you're experiencing this issue.

GitHub is a software hosting service, we host millions of projects, so probably every phone and computer on the planet has software on it that is also hosted here.

We're not able to provide support or answers for those individual projects. GitHub Support can only provide support for GitHub itself.

I'm sorry we can't help here but we'd recommend contacting a local IT specialist for assistance or device's community. They should be able to help you with these applications!
